Fixes issue 2498: Ensure no trailing slash in node addresses

hewison-chris opened the pull request:
Prevent potential duplicate registered nodes where address are included with and without a trailing slash.

linked0 reviewed (approved):
LGTM

Geod24 commented on source/agora/network/Manager.d:
```diff
@@ -786,6 +786,9 @@ public class NetworkManager
     /// Ditto
     private void addAddress (Address address)
     {
+        // Ensure we have a trailing slash for address
+        if (address[$ - 1] != '/')
+            address ~= '/';
```
Hum, I'm not sure that'd work. Address could technically be an IP, couldn't it ?

hewison-chris replied:
```
node-6_1  | 2021-09-16 08:44:11,239 Info [agora.network.Manager] - Found new FullNode: https://node-0:1735/
node-6_1  | 2021-09-16 08:44:11,240 Info [agora.network.Client] - Response of https://node-0:1735//register_listener : HTTP/1.1 404 Not Found
```

hewison-chris replied:
The CI passed yet the clients failed to register. :thinking:

hewison-chris replied:
> Hum, I'm not sure that'd work. Address could technically be an IP, couldn't it ?

Yes and we also add a slash in REST endpoints. I think it makes more sense to store in `Network.Manager` without the slash.

hewison-chris committed:
Ensure no trailing slashes in node address

As addresses can be configured to include a slash at the end. If it does
we strip it off. This ensures that when we add a rest path after the
address we do not end up with a double slash in the path.

hewison-chris commented:
This has been addressed by other changes so will close this PR.

Geod24 commented:
Which change ? I don't remember any that fixed this. AFAIR we only discussed going with another approach.

hewison-chris commented:
My mistake I'll re open this issue.

Geod24 commented on source/agora/network/Manager.d:
```diff
@@ -786,6 +786,9 @@ public class NetworkManager
     {
+        // Ensure we do not have a trailing slash for address
+        if (address[$ - 1] == '/')
+            address.length -= 1;
```
```suggestion
        // Ensure we do not have trailing slashes for address
        while (address.length && address[$ - 1] == '/')
            address.length -= 1;
        if (!address.length) return;
```

Let's go with this for the time being to fix the bug in production.

hewison-chris replied:
Rebased and added your suggestion.

kchulj reviewed (approved):
LGTM
